src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -4441,12 +4480,16 @@ private void calculateToFromStreams(Collection<Token> newTokens, List<String> ke
                     Multimap<InetAddress, Range<Token>> workMap = RangeStreamer.getWorkMap(rangesToFetchWithPreferredEndpoints, keyspace, FailureDetector.instance, useStrictConsistency);
                     for (InetAddress address : workMap.keySet())
                     {
-                        logger.debug("Will request range {} of keyspace {} from endpoint {}", workMap.get(address), keyspace, address);
+                        logger.debug("Will request range {} of keyspace {} from endpoint {}",
+                                     SafeArg.of("range", workMap.get(address)),
+                                     SafeArg.of("keyspace", keyspace),
+                                     SafeArg.of("endpoint", address));
                         InetAddress preferred = SystemKeyspace.getPreferredIP(address);
                         streamPlan.requestRanges(address, preferred, keyspace, workMap.get(address));
                     }
 
-                    logger.debug("Keyspace {}: work map {}.", keyspace, workMap);
+                    // This log line seems unnecessary given the above loop...
```

Comment:
maybe, but I do think in grafana it's easier to analyze a single log line instead of combining multiple, especially when multiple nodes are reporting the logs simultaneously

Reply:
This is true as long as `workMap` doesn't get so large that we can't actually ingest the log. I don't have any evidence to suggest that's the case, but I don't know that I necessarily agree with the "single log > multi log" statement... filtering on tags like `host` exists for a reason 😛